internal/yunikorn/event_handler.go

```diff
@@ -83,32 +83,48 @@ func (s *Service) handleAppEvent(ctx context.Context, ev *si.EventRecord) {
 func (s *Service) handleQueueEvent(ctx context.Context, ev *si.EventRecord) {
 	logger := log.FromContext(ctx)
 	logger.Debugf("adding queue event to accumulator: %v", ev)
-	s.queueEventAccumulator.add(ev)
-}
-
-func (s *Service) handleQueueEvents(ctx context.Context, events []*si.EventRecord) {
-	logger := log.FromContext(ctx)
 
-	logger.Debug("Processing queue events")
-	for _, event := range events {
-		logger.Debugf("Event: %v", event)
+	var daoQueue dao.PartitionQueueDAOInfo
+	if err := json.Unmarshal([]byte(ev.GetState()), &daoQueue); err != nil {
+		logger.Errorw("Failed to unmarshal queue state from event", "error", err)
+		return
 	}
 
-	s.handleQueueAddEvent(ctx)
-	logger.Debug("Finished processing queue events")
-}
+	isNew := ev.GetEventChangeType() == si.EventRecord_ADD &&
+		(ev.GetEventChangeDetail() == si.EventRecord_DETAILS_NONE || ev.GetEventChangeDetail() == si.EventRecord_QUEUE_DYNAMIC)
 
-func (s *Service) handleQueueAddEvent(ctx context.Context) {
-	logger := log.FromContext(ctx)
+	var queue *model.Queue
+	if isNew {
+		s.partitionAccumulator.add(ev)
+		queue = &model.Queue{
+			Metadata: model.Metadata{
+				ID:            ulid.Make().String(),
+				CreatedAtNano: ev.TimestampNano,
+			},
+			PartitionQueueDAOInfo: daoQueue,
+		}
+
+		if err := s.repo.InsertQueue(ctx, queue); err != nil {
+			logger.Errorf("could not insert queue: %v", err)
+			return
+		}
```

Comment:
It appears that the following insert will always fail due to a `NOT NULL` constraint on the `partition` field in the database (`partition TEXT NOT NULL CHECK (partition <> '')`).

YuniKorn does not provide a `partition` value for child queues, which results in an empty string (`""`) for the `partition` field when we receive events `state`. As a result, inserting these records will cause a DB error.

Reply:
For the initial sync it works because we update the `Partition` with the parent partition for all children in `flattenQueues()` method